actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -569,7 +580,7 @@ def parse(name, signed_message)
         end
 
         def commit(options)
-          options[:value] = @verifier.generate(serialize(options[:value]))
+          options[:value] = @verifier.generate(serialize(options[:value]), expires_at: options[:expires_at], expires_in: options[:expires_in])
 
```

Comment:
@kaspth  :smile: 
Does the following sound reasonable?
Set the verifier to use the serializer used by cookies instead of NullSerializer, this way the wrapped message could be serialized with the cookie_serializer.

To preserve backwards compatibility we could have a switch for it.